codetransformer/transformers/constants.py

```diff
@@ -38,7 +75,17 @@ def __init__(self, *args, **kwargs):
                 raise TypeError('Duplicate keys: {!r}'.format(overlap))
             constnames.update(kwargs)
 
-    @pattern(LOAD_NAME | LOAD_GLOBAL)
+    def transform(self, code, **kwargs):
+        overlap = self._constnames.keys() & set(code.argnames)
+        if overlap:
+            raise SyntaxError(
+                'argument names overlap with constant names: %r' % overlap,
+            )
+        return super().transform(code, **kwargs)
+
+    @pattern(
+        LOAD_NAME | LOAD_GLOBAL | LOAD_DEREF | LOAD_CLASSDEREF | LOAD_FAST,
```

Comment:
I could either not handle this case, or dispatch on load_fast and check the name to raise an assertionerror, your call. I am sort of in support of not adding an extra function just to make this assertion but your call.